issue99_test.go

```diff
@@ -0,0 +1,78 @@
+package zmq4
+
+import (
+	"context"
+	"sync"
+	"testing"
+	"time"
+)
+
+var (
+	wg     sync.WaitGroup
+	outMsg Msg
+	inMsg  Msg
+)
+
+func requester(t *testing.T) {
+
+	req := NewReq(context.Background())
+	defer req.Close()
+
+	err := req.Dial("tcp://localhost:5559")
+	if err != nil {
+		wg.Done()
```

Comment:
please remove this, and all other occurrences.
and add a single `defer wg.Done()` at the top of the function.